src/plonkish/backend/halo2.rs

```diff
@@ -671,39 +672,73 @@ fn assign_witness<F: PrimeField + From<u64>>(
     }
 }
 
-pub trait PlonkishHalo2<F: PrimeField, W, WG: Halo2WitnessGenerator<F, W>> {
+#[allow(private_bounds)]
+pub trait PlonkishHalo2<W, WG: Halo2WitnessGenerator<Fr, W>>: Halo2Compilable<W, WG> {
```

Comment:
@leolara I need some advice here on how to better deal with the common functionality while creating the provers. The only difference between supercircuit and single circuit is in the compilation, so I offloaded it into a `Halo2Compilable` trait. I want it to stay private because it's only used in our halo2 backend. At the same time, it should be shared between both implementations of `PlonkishHalo2`. When I make it a super trait of `PlonkishHalo2` here the clippy gives warning that `Halo2Compilable` is more private than `PlonkishHalo2` which is my intention because I don't want to expose the compilation to a Chiquito API consumer. Do you think there's a better way to do that while not repeating the code?

Reply:
Does this compile, that PlonkishHalo2 is public descendant from a private trait? Interesting

Reply:
It does :)

Reply:
This looks like a good solution

Reply:
Perhaps, it would be even more interesting a blanket trait impl

```
impl<T: Halo2Compilable> PlonkishHalo2 for T {
    fn create_halo2_prover ....
}
```

Reply:
That way PlonkishHalo2 wouldn't have to inherit from the private Halo2Compilable